addons/mail/models/mail_thread_main_attachment.py

```diff
@@ -23,11 +23,10 @@ def _message_post_after_hook(self, message, msg_values):
 
     def _message_set_main_attachment_id(self, attachment_ids):
         if attachment_ids and not self.message_main_attachment_id:
-            all_attachments = self.env['ir.attachment'].browse(attachment_ids)
-            prioritary_attachments = all_attachments.filtered(lambda x: x.mimetype.endswith('pdf')) \
-                                     or all_attachments.filtered(lambda x: x.mimetype.startswith('image')) \
-                                     or all_attachments
-            self.sudo().with_context(tracking_disable=True).message_main_attachment_id = prioritary_attachments[0].id
+            self.sudo().with_context(tracking_disable=True).message_main_attachment_id = min(
```

Comment:
Would add a comment to clarify it takes the first pdf if available, then the next image if available, then the first attachment

Reply:
Yes indeed, good idea